jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/JibPluginConfiguration.java

```diff
@@ -146,6 +147,16 @@ public PermissionConfiguration() {}
     public Optional<String> getArchitectureName() {
       return Optional.ofNullable(architecture);
     }
+
+    private static PlatformParameters ofString(String osArchitecture) {
+      if (osArchitecture == null || !osArchitecture.matches("[^/]+/[^/]+")) {
+        throw new IllegalArgumentException("osArchitecture must be of form os/architecture");
+      }
+      PlatformParameters platformParameters = new PlatformParameters();
+      platformParameters.os = osArchitecture.split("/")[0].trim();
+      platformParameters.architecture = osArchitecture.split("/")[1].trim();
+      return platformParameters;
```

Comment:
We use NullAway. The null check is unnecessary. And I don't like allowing whitespace and using `.trim()`. Let's just say `[^/ ]+/[^/ ]+`. And then, I think we can make this simpler:

```java
      Matcher matcher = Pattern.compile("([^/ ]+)/([^/ ]+)").matcher(osArchitecture);
      if (!matcher.matches()) {
        throw ...
      }
      PlatformParameters platformParameters = new PlatformParameters();
      platformParameters.os = matcher.group(1);
      platformParameters.architecture = matcher.group(2);
      return platformParameters;
```

And let's move this method to the top of the class (sorry, our repo style):
```java

public static class PlatformParameters implements PlatformConfiguration {

  private static PlatformParameters ofString(String osArchitecture) { ...
```